packages/bot-web-ui/src/app/app.jsx

```diff
@@ -64,16 +66,22 @@ const App = ({ passthrough }) => {
     ) : (
         <MobxContentProvider store={root_store_instance.current}>
             <div className='bot'>
-                <BotNotificationMessages />
-                <NetworkToastPopup />
-                <Toolbar />
-                <MainContent />
-                <RunPanel />
-                <QuickStrategy />
-                <BotFooterExtensions />
-                <Audio />
-                <RoutePromptDialog />
-                <BlocklyLoading />
+                {show_dashboard ? (
+                    <Dashboard />
+                ) : (
+                    <>
+                        <BotNotificationMessages />
+                        <NetworkToastPopup />
+                        <Toolbar />
+                        <MainContent />
```

Comment:
If show_dashboard === true, we don't have `scratch_div` so we may get unexpected errors on mounting the blockly workspace. I suggest you update useEffect and check `show_dashboard`